tensorboard/plugins/beholder/tf_beholder_dashboard/tf-beholder-info.html

```diff
@@ -49,39 +49,81 @@
     "use strict";
     (function() {
 
-    const INFO_ROUTE = tf_backend.getRouter().pluginRoute('beholder', '/section-info')
+    const INFO_ROUTE = tf_backend.getRouter().pluginRoute('beholder', '/section-info');
 
     Polymer({
       is: "tf-beholder-info",
       properties: {
-        _requestManager: {
-          type: Object,
-          value: () => new tf_backend.RequestManager(),
+        fps: {  // Not actually FPS!
+          type: Number,
+          value: 10,
+        },
+
+        xhrTimeout: {
+          type: Number,
+          value: 10000,
         },
 
         items: {
           type: Array,
           value: [{name:'Loading...'}],
         },
+
+        _xhr: Object,
+        _timer: Object,
       },
 
-      _reloaditems() {
-        window.setTimeout(function(){
-          this._requestManager.request(INFO_ROUTE).then(function(response){
-            this.splice('items', 0)
+      attached() {
+        this._load();
+      },
 
-            for (var item of response) {
-             this.push('items', item)
-            }
-          }.bind(this))
+      detached() {
+        this._clear();
+      },
+
+      _load() {
+        this._clear();
+        this._xhr = new XMLHttpRequest();
+        this._xhr.open('GET', INFO_ROUTE, true);
+        this._xhr.timeout = this.xhrTimeout;
+        this._xhr.onload = this._onLoad.bind(this);
+        this._xhr.onerror = this._retry.bind(this, this._getSleep());
+        this._xhr.ontimeout = this._retry.bind(this, 1);
+        this._xhr.send(null);
+      },
+
+      _onLoad() {
+        if (this._xhr.status == 200) {
+          const response = JSON.parse(this._xhr.responseText);
+          this.splice('items', 0);
+          for (var item of response) {
+            this.push('items', item);
+          }
+        }
+        this._retry(this._getSleep());
+      },
+
+      _retry(delay) {
+        this._timer = window.setTimeout(this._load.bind(this), delay);
+      },
+
+      _getSleep() {
+        return 1000 / (this.fps === 0 ? 1 : this.fps);
```

Comment:
When fps is 0, the section info should not be refreshed.  So changing fps to 0 should result in the same cessation of XHRs and timers that result from detached().  Otherwise XHRs will continue to be sent, even though there is no new data to report.

This is also true of tf-beholder-video, which continues pinging even when FPS is set to 0 in the containing dashboard.

Reply:
I solved this by setting the FPS slider minimum to 1. Please note that, after this change, FPS is now "FPS". I'm not too concerned here. I disagree with the FPS slider. In an ideal world, this would websocket push (not poll). Then it becomes an operations concern if pushes happen more frequently than either side can handle.

Reply:
Please revert that "solution" - by removing the ability to set FPS to 0, you're removing a key feature that has been part of Beholder since Chris wrote it.  It's important both to the UI - it lets you pause the video updates so that you have time to inspect the visualization and section values in detail, which is basically impossible at a 1 second update interval - and also to training performance, since by setting FPS to 0 it tells the actual Beholder code in the model to skip frame generation to reduce overhead during training.  The feature is already documented in [his readme](https://github.com/chrisranderson/beholder/blame/06a0cd4ffbda55e79a46ce98cd82ecc851523038/README.md#L129), [our readme](https://github.com/tensorflow/tensorboard/blame/70aece3b60800d872cf60cb0f4ab9a5dc1f94ed7/tensorboard/plugins/beholder/README.md#L134-L135), and in [his video](https://www.youtube.com/watch?v=06HjEr0OX5k&t=114).

We shouldn't take away a valuable feature just because in an ideal world we would reimplement it using websockets (and as a aside, I still think you would want an FPS control for UX reasons as well as to be able to throttle down the generation overhead on the model training side, so I don't actually see how that would change things).

If you're not interested in fixing the code to avoid XHRs when FPS is set to 0, we can leave it as is and I will fix it post-release.